src/hubverse_infrastructure/shared/hubverse_transforms.py

```diff
@@ -0,0 +1,225 @@
+"""Create the AWS infrastructure needed to run Hubverse model-output transformations as Lambda functions."""
+import io
+from zipfile import ZipFile
+
+import boto3
+import pulumi
+import pulumi_aws as aws
+from botocore.exceptions import ClientError
+from cloudpathlib import CloudPath
+from pulumi import ResourceOptions  # type: ignore
+
+
+def create_bucket(bucket_name: str) -> aws.s3.BucketV2:
+    """
+    Create the S3 bucket used to store shared Hubverse assets and give the AWS Lambda service read access to it.
+    """
+
+    hubverse_asset_bucket = aws.s3.BucketV2(bucket_name, bucket=bucket_name, tags={"hub": "hubverse"})
+
+    # Create an S3 policy that allows the AWS Lambda service to read from the bucket
+    # (this is a permissive read policy because it applies to any lambda function, but the Hubverse
+    # deals in open source data and code, so we can leave keep it open for simplicity).
+    s3_policy_document = aws.iam.get_policy_document(
+        statements=[
+            aws.iam.GetPolicyDocumentStatementArgs(
+                sid="ReadAssetBucket",
+                actions=[
+                    "s3:GetObject",
+                    "s3:GetObjectAcl",
+                ],
+                principals=[
+                    aws.iam.GetPolicyDocumentStatementPrincipalArgs(
+                        type="Service", identifiers=["lambda.amazonaws.com"]
```

Comment:
Here and below, the specified service `principal` means that the allowed actions can be done by any lambda function in the Hubverse's AWS account